qiskit_nature/second_q/formats/fcidump/fcidump.py

```diff
@@ -0,0 +1,131 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""FCIDump"""
+
+from __future__ import annotations
+
+from typing import List
+from dataclasses import dataclass
+from pathlib import Path
+import numpy as np
+
+from qiskit_nature import QiskitNatureError
+
+
+from .dumper import _dump_1e_ints, _dump_2e_ints, _write_to_outfile
+from .parser import _parse
+
+
+@dataclass
+class FCIDump:
+    """
+    Qiskit Nature dataclass for reading an FCIDump file.
+
+    The FCIDump format is partially defined in Knowles1989.
+
+    References:
+        Knowles1989: Peter J. Knowles, Nicholas C. Handy,
+            A determinant based full configuration interaction program,
+            Computer Physics Communications, Volume 54, Issue 1, 1989, Pages 75-83,
+            ISSN 0010-4655, https://doi.org/10.1016/0010-4655(89)90033-7.
+    """
+
+    hij: np.ndarray
+    """The alpha 1-electron integrals."""
+    hij_b: np.ndarray | None
+    """The beta 1-electron integrals."""
+    hijkl: np.ndarray
+    """The alpha/alpha 2-electron integrals."""
+    hijkl_ba: np.ndarray | None
+    """The beta/alpha 2-electron integrals."""
+    hijkl_bb: np.ndarray | None
+    """The beta/beta 2-electron integrals."""
+    multiplicity: int
+    """The multiplicity."""
+    num_electrons: int
+    """The number of electrons."""
+    num_orbitals: int
+    """The number of orbitals."""
+    constant_energy: float | None
+    """The constant energy comprising nuclear repulsion energy and inactive energies"""
+    orbsym: List[str] | None
+    """A list of spatial symmetries of the orbitals."""
+    isym: int
+    """The spatial symmetry of the wave function."""
+
+    @classmethod
+    def from_file(cls, fcidump: str | Path) -> FCIDump:
+        """Constructs an FCIDump object from a file."""
+        data = _parse(fcidump if isinstance(fcidump, Path) else Path(fcidump))
+        return cls(
```

Comment:
Seeing this now, would it not make more sense, if the `_parse` method were to construct the `FCIDump` dataclass object directly? That might get slightly tricky with import cyclicity, but it would avoid the packing/unpacking of this `data` dictionary, which we are doing here right now